lib/data/file_manager/file_manager.dart

```diff
@@ -390,18 +393,19 @@ class FileManager {
         .map((FileSystemEntity entity) {
           String filePath = entity.path.substring(documentsDirectory.length);
 
-          // remove extension
-          if (filePath.endsWith(Editor.extension)) {
-            filePath = filePath.substring(
-                0, filePath.length - Editor.extension.length);
-          } else if (filePath.endsWith(Editor.extensionOldJson)) {
-            filePath = filePath.substring(
-                0, filePath.length - Editor.extensionOldJson.length);
+          if (removeExtension){
+            // remove extension
+            if (filePath.endsWith(Editor.extension)) {
+              filePath = filePath.substring(
+                  0, filePath.length - Editor.extension.length);
+            } else if (filePath.endsWith(Editor.extensionOldJson)) {
+              filePath = filePath.substring(
+                  0, filePath.length - Editor.extensionOldJson.length);
+            }
+
+            if (Editor.isReservedPath(filePath))
```

Comment:
I see that you've moved this into the `if (removeExtension)` block.
I think it'd make more sense if we add another parameter for `includeReservedPaths` rather than tying it into an unrelated flag

Reply:
I suggested some code, which worked for me. I am not familiar with programming dart and flutter applications so my code is not optimal (I am FORTRAN programmer of scientific computations). 

Please modify the code completely according to your ideas. I don't stand by what I wrote at all. I'm not used to collaborating on code creation, so I would ask you to process everything yourself.

Reply:
I will try to apply your suggestions and commit it. Hope it saves you some time